src/Sulu/Bundle/MediaBundle/Tests/Unit/Media/Storage/FlysystemStorageTest.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\Tests\Unit\Media\Storage;
+
+use League\Flysystem\FilesystemAdapter;
+use League\Flysystem\FilesystemOperator;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Prophecy\Prophecy\ObjectProphecy;
+use Sulu\Bundle\MediaBundle\Media\Storage\FlysystemStorage;
+
+class FlysystemStorageTest extends TestCase
+{
+    use ProphecyTrait;
+
+    private FlysystemStorage $flysystemStorage;
+
+    /**
+     * @var ObjectProphecy<FilesystemOperator>
+     */
+    private $flysystem;
+
+    /**
+     * @var ObjectProphecy<FilesystemAdapter>
+     */
+    private ObjectProphecy $flysystemAdapter;
+
+    protected function setUp(): void
+    {
+        $this->flysystem = $this->prophesize(FilesystemOperator::class);
+        $this->flysystemAdapter = $this->prophesize(FilesystemAdapter::class);
+
+        $this->flysystemStorage = new FlysystemStorage(
+            $this->flysystem->reveal(),
+            $this->flysystemAdapter->reveal(),
+            100
+        );
```

Comment:
Would not mock here anything just use the in memory adapter: https://github.com/thephpleague/flysystem-memory

Reply:
Done